src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -284,91 +275,149 @@ runTestCase( const std::shared_ptr<xrt_core::device>& _dev, const std::string& p
 
     // Validate the legacy names
     // If no legacy name exists use the passed in test name
-    std::string test_name = py;
-    if (test_map.find(py) != test_map.end())
-      test_name = test_map.find(py)->second;
-
-    // Parse if the file exists here
-    std::string  xrtTestCasePath = "/opt/xilinx/xrt/test/" + test_name;
-    boost::filesystem::path xrt_path(xrtTestCasePath);
-    if (!boost::filesystem::exists(xrt_path)) {
-      logger(_ptTest, "Error", boost::str(boost::format("Failed to find %s") % xrtTestCasePath));
-      logger(_ptTest, "Error", "Please check if the platform package is installed correctly");
-      _ptTest.put("status", test_token_failed);
-      return;
-    }
+    std::string test_name = file;
+    if (test_map.find(file) != test_map.end())
+      test_name = test_map.find(file)->second;
 
-    // log testcase path for debugging purposes
-    logger(_ptTest, "Testcase", xrtTestCasePath);
+    xrtTestCasePath = "/opt/xilinx/xrt/test/" + test_name;
+  } else {
+    xrtTestCasePath = "/opt/xilinx/xrt/test/" + file;
+  }
+  // Parse if the file exists here
+  boost::filesystem::path xrt_path(xrtTestCasePath);
+  if (!boost::filesystem::exists(xrt_path)) {
+    logger(_ptTest, "Error", boost::str(boost::format("Failed to find %s") % xrtTestCasePath));
+    logger(_ptTest, "Error", "Please check if the platform package is installed correctly");
+    _ptTest.put("status", test_token_failed);
+    throw xrt_core::error(std::errc::operation_canceled);
+  }
+  return xrtTestCasePath;
+}
 
-    std::vector<std::string> args = { "-p", test_dir,
-                                      "-d", xrt_core::query::pcie_bdf::to_string(xrt_core::device_query<xrt_core::query::pcie_bdf>(_dev)) };
-    try {
-      int exit_code = XBU::runScript("sh", xrtTestCasePath, args, "Running Test", MAX_TEST_DURATION, os_stdout, os_stderr);
-      if (exit_code == EOPNOTSUPP) {
-        _ptTest.put("status", test_token_skipped);
-      }
-      else if (exit_code == EXIT_SUCCESS) {
+/*
+ * Spawn a testcase process.
+ */
+bool
```

Comment:
Why does this method have a return type of bool.  Shouldn't it be void and an exception thrown instead?